Tests/SwiftSyntaxTestSupportTest/IncrementalParseTestUtilsTest.swift

```diff
@@ -0,0 +1,130 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import SwiftSyntax
+import _SwiftSyntaxTestSupport
+
+public class IncrementalParseTestUtilsTest: XCTestCase {
+  public func testReuseRangeBeforeEditedArea() {
+    let source =
+      """
+      ➡️struct B {}⬅️
+      struct A⏩️⏸️A⏪️ {}
+      """
+
+    let expectedOriginalSource =
+      """
+      struct B {}
+      struct A {}
+      """
+
+    let expectedEditedSource =
+      """
+      struct B {}
+      struct AA {}
+      """
+
+    let (originalSource, editedSource, edits, reusedRange) = extractMarkersForIncrementParseTest(source)
+    XCTAssertEqual(originalSource, expectedOriginalSource)
+    XCTAssertEqual(editedSource, expectedEditedSource)
+    XCTAssertEqual(
+      edits.edits,
+      [
+        SourceEdit(offset: 20, length: 0, replacementLength: 1)
+      ]
+    )
+    XCTAssertEqual(
+      reusedRange,
+      [
+        ByteSourceRange(offset: 0, length: 11)
+      ]
+    )
+  }
+
+  public func testReuseRangeAfterEditedArea() {
+    let source =
+      """
+      struct A⏩️⏸️A⏪️ {}
+      ➡️struct B {}⬅️
+      """
+
+    let expectedOriginalSource =
+      """
+      struct A {}
+      struct B {}
+      """
+
+    let expectedEditedSource =
+      """
+      struct AA {}
+      struct B {}
+      """
+
+    let (originalSource, editedSource, edits, reusedRange) = extractMarkersForIncrementParseTest(source)
+    XCTAssertEqual(originalSource, expectedOriginalSource)
+    XCTAssertEqual(editedSource, expectedEditedSource)
+    XCTAssertEqual(
+      edits.edits,
+      [
+        SourceEdit(offset: 8, length: 0, replacementLength: 1)
+      ]
+    )
+    XCTAssertEqual(
+      reusedRange,
+      [
+        ByteSourceRange(offset: 12, length: 11)
+      ]
+    )
+  }
+
+  public func testReuseRangeInsideMultipleEdits() {
+    let source =
+      """
+      ⏩️⏸️class foo {⏪️
+        ➡️let value1 = 👨‍👩‍👧‍👧⬅️
+      ⏩️⏸️}⏪️
+      """
```

Comment:
I feel like we are getting into marker hell here with the two different kinds of markers. Just an idea: What do you think about adding a `reused: String...` parameter to `assertIncrementalParse`. In this case you would be passing `let value1 = 👨‍👩‍👧‍👧` to that parameter to assert that a node with that source text has been reused. We do something similar with the `highlight` parameter on `DiagnosticSpec`.

Reply:
Yeah, I also added a `ReusedNodeSpec` in the new commit, which makes the usage of this function very much like `assertParse`.

Reply:
Thanks. It’s looking very good